torch/_subclasses/meta_utils.py

```diff
@@ -225,6 +225,41 @@ def describe_tensor(self, t: torch.Tensor, recurse: bool = True):
             # sym_sizes_strides_storage_offset (empty_create)
             stride = t.stride()
 
+        # NB: this technically should refer to functorch unwrapped tensor, but
+        # I am (perhaps abusively) using it to store both the functorch and
+        # non-functorch functional tensor
+        unwrapped = None
+        autograd_meta_from = None
+        current_level = None
+        if is_batchedtensor_v or is_gradtrackingtensor_v:
+            unwrapped = self.describe_tensor(get_unwrapped(t))
+        # xla and lazy tensors present as functional tensors, but we want them
+        # to be handled specially
+        elif is_functional and t.device.type not in ("xla", "lazy"):
+            if t._is_view():
+                raise RuntimeError(
+                    "Cannot safely fakify a view because this process drops the view information right now."
+                )
+            if not is_functorch_wrapped:
+                torch._sync(t)
+                unwrapped = self.describe_tensor(torch._from_functional_tensor(t))
+                autograd_meta_from = t
+            else:
+                reapply_views = torch._C._functionalization_reapply_views_tls()
+                # NB: has side effects!
+                unwrapped = self.describe_tensor(
+                    _unwrap_functional_tensor(t, reapply_views)
+                )
+                # TODO: It's pretty suspicious that functional tensors don't have
+                # valid level and thus we just grab whatever the current level
+                # is
+                current_level = torch._C._functorch.current_level()
+
+        functorch_stack = None
+        if is_functorch_wrapped:
+            with torch._functorch.pyfunctorch.temporarily_clear_interpreter_stack() as functorch_stack:
```

Comment:
To check, we're just using this to retrieve the functorch stack? (a helper to retrieve the stack might be better?)

Reply:
Yes. If you want me to, I can unify the interpreter stack and dynamic stacks (as the dynamic stack as half the API I need and interpreter stack has the other half), but I'll need some guidance on what you want the end state to be.

Reply:
Let's leave it for now, I don't want to figure out unifying those right now lol